include/eve/module/algo/algo/transform_copy_if.hpp

```diff
@@ -0,0 +1,168 @@
+#pragma once
+
+#include <eve/module/algo/algo/common_forceinline_lambdas.hpp>
+#include <eve/module/algo/algo/traits.hpp>
+#include <eve/module/algo/algo/two_stage_iteration.hpp>
+#include <eve/module/algo/algo/views/convert.hpp>
+#include <eve/module/core.hpp>
+
+namespace eve::algo
+{
+
+  namespace detail
+  {
+    template<typename TraitsSupport> struct transform_copy_if_ : TraitsSupport
+    {
+      template<typename O, typename F> struct delegate
+      {
+        O of;
+        O ol;
+        F func;
+
+        EVE_FORCEINLINE
+        bool tail(auto f, auto ignore)
+        {
+          auto out_ignore   = eve::keep_first(std::min(ol - of, eve::iterator_cardinal_v<O>));
+          auto loaded       = eve::load[ignore](f);
+          auto [vals, mask] = func(loaded);
+          using out_t = eve::element_type_t<decltype(vals)>;
+          auto cvt_and_store_it = views::convert(of, eve::as<out_t>{});
+          auto write_end = eve::compress_store[eve::unsafe][ignore][out_ignore](vals, mask, cvt_and_store_it);
+          of += write_end - cvt_and_store_it;
+          return of == ol;
+        }
+
+        EVE_FORCEINLINE
+        std::ptrdiff_t left_for_stage1() const
+        {
+          return ol - of;
+        }
+
+        EVE_FORCEINLINE
+        bool step_1(auto f)
+        {
+          auto loaded       = eve::load(f);
+          auto [vals, mask] = func(loaded);
+          using out_t = eve::element_type_t<decltype(vals)>;
+          auto cvt_and_store_it = views::convert(of, eve::as<out_t>{});
+          auto write_end = eve::compress_store[eve::unsafe](vals, mask, cvt_and_store_it);
+          of += write_end - cvt_and_store_it;
+          return false;
+        }
+
+        EVE_FORCEINLINE
+        bool step_2(auto f)
+        {
+          // ol - of < cardinal
+          auto out_ignore   = eve::keep_first(ol - of);
+          auto loaded       = eve::load(f);
+          auto [vals, mask] = func(loaded);
+          using out_t = eve::element_type_t<decltype(vals)>;
+          auto cvt_and_store_it = views::convert(of, eve::as<out_t>{});
+          auto write_end = eve::compress_store[eve::unsafe][eve::ignore_none][out_ignore](vals, mask, cvt_and_store_it);
+          of += write_end - cvt_and_store_it;
+          return of == ol;
+        }
+      };
+
+      template<relaxed_range In, relaxed_range Out, typename Func>
+      EVE_FORCEINLINE
+      auto operator()(In&& in, Out&& out, Func func) const -> unaligned_iterator_t<Out>
+      {
+        if( in.begin() == in.end() || out.begin() == out.end() )
+          return out.begin();
+
+        auto [processed_in, processed_out] = temporary_preprocess_ranges_hack(
+          TraitsSupport::get_traits(), in, out);
+
+        auto iteration = two_stage_iteration(processed_in.traits(), processed_in.begin(), processed_in.end());
+
+        auto of = unalign(processed_out.begin());
+        auto ol = unalign(processed_out.end());
```

Comment:
I don't like that we do conversions in all callbacks. 

Let's figure out conversion here once or - hide it in some function.

Reply:
Try this one maybe please? I think it'd be cleaner.